src/prefect/utilities/collections.py

```diff
@@ -258,6 +259,7 @@ def visit_nested(expr):
             expr,
             visit_fn=visit_fn,
             return_data=return_data,
+            max_depth=max_depth - 1,
```

Comment:
I'm confused by this, if we default `max_depth` to `-1` and then we subtract `1` when recursing, how do we ever end up with a `max_depth` of `0` ?

Reply:
The new docstring should be helpful. Negative values indicate infinite recursion.